assets/tests/TaskSchemaValidationService.js

```diff
@@ -0,0 +1,262 @@
+        for (var i = 0; i < performanceTests.length; i++) {
+          var test = performanceTests[i];
+          var inputDataAtom = test.getInputDataAtom();
+          if (!inputDataAtom) {
+            return false;
+          }
+          if (!(typeof inputDataAtom === 'string' ||
+            inputDataAtom instanceof String)) {
+            return false;
+          }
+        }
+        return true;
+      },
+      verifyPerformanceTestsHaveNonEmptyTransformationFunctionName: function(
+        task, auxiliaryCode) {
+        var performanceTests = task.getPerformanceTests();
+        for (var i = 0; i < performanceTests.length; i++) {
+          var test = performanceTests[i];
+          var transformationFunctionName = test.getTransformationFunctionName();
+          if (!transformationFunctionName) {
+            return false;
+          }
+          if (!(typeof transformationFunctionName === 'string' ||
+            transformationFunctionName instanceof String)) {
+            return false;
+          }
+          var auxiliaryCodeClassName = 'AuxiliaryCode.';
+          if (transformationFunctionName.startsWith(auxiliaryCodeClassName)) {
+            var functionName = transformationFunctionName.substring(
+              auxiliaryCodeClassName.length);
+            return auxiliaryCode.indexOf(functionName) !== -1;
+          }
+          var systemCodeClassName = 'System.';
+          if (transformationFunctionName.startsWith(systemCodeClassName)) {
+            // TODO(eyurko): Make system code accessible by testing framework.
+            return true;
+          }
+          return false;
+        }
+        return true;
+      },
+      verifyPerformanceTestsHaveLinearExpectedPerformance: function(
+        task) {
+        var performanceTests = task.getPerformanceTests();
+        for (var i = 0; i < performanceTests.length; i++) {
+          var test = performanceTests[i];
+          // TODO(eyurko): Update this once we support nonlinear runtimes.
+          if (test.getExpectedPerformance() !== 'linear') {
+            return false;
+          }
+        }
+        return true;
+      },
+      verifyPerformanceTestsHaveEvaluationFunctionName: function(
+        task, auxiliaryCode, starterCode) {
+        var performanceTests = task.getPerformanceTests();
+        for (var i = 0; i < performanceTests.length; i++) {
+          var test = performanceTests[i];
+          var evaluationFunctionName = test.getEvaluationFunctionName();
+          if (evaluationFunctionName === null) {
+            return false;
+          }
+          if (!(typeof evaluationFunctionName === 'string' ||
+            evaluationFunctionName instanceof String)) {
+            return false;
+          }
+          var auxiliaryCodeClassName = 'AuxiliaryCode.';
+          if (evaluationFunctionName.startsWith(auxiliaryCodeClassName)) {
+            var functionName = evaluationFunctionName.substring(
```

Comment:
I feel like you _might_ want to break out a function like `checkIfFunctionExistsInCode(functionName, code), since you seem to be doing this quite a bit. You can just make it a private function in this service.